### DISCUSSION TITLE
How to use Dependency clock with swift-async-algorithms

### DISCUSSION THREAD

**Discussion by fatbobman** [pointfreeco/swift-composable-architecture#1626] [General]
I want to use swift-async-algorithms in Reducer to make some adjustments to async sequences.
However, the default clock Key provided by Dependency cannot be used directly in the method, prompt: `Type 'any Clock<Duration>' cannot conform to 'Clock'`

Excuse me, how to combine the two together?

```swift
 @Dependency(\.continuousClock)
 var clock

// Type 'any Clock<Duration>' cannot conform to 'Clock'
for await objects in await fetcher().stream.debounce(for: .seconds(0.3), clock: clock) {
        send(.objectsUpdateResponse(.values(objects)))
 }
```

**Comment by mbrandonw** [comment]
Huh, I thought this would have worked, but sadly it seems that Swift's existentials are not powerful enough for this. This is a pretty big bummer.

I think the only way to work around is to use the `_AnyClock` that ships with swift-clocks:

```swift
for await objects in await fetcher().stream.debounce(for: .seconds(0.3), clock: _AnyClock(clock)) {
  …
}
```

We thought we could keep `_AnyClock` as an internal detail but maybe we need to publicize it. And maybe `@Dependency(\.continuousClock)` needs to be an `AnyClock` instead of an `any Clock`.

**Reply by fatbobman** [reply]
Thanks a lot, after using `clock: _AnyClock(self.clock)` the code compiles.

**Reply by stephencelis** [reply]
@fatbobman We decided to make this type more publicly usable: https://github.com/pointfreeco/swift-clocks/pull/11

You'll want to be aware of this when you upgrade in the future, as you'll want to rename `_AnyClock` to `AnyClock`.

**Reply by fatbobman** [reply]
Thanks!! Really good news
